unionml/model.py

```diff
@@ -109,6 +109,8 @@ class in sklearn, ``Module`` in pytorch, etc.
         self._train_task = None
         self._predict_task = None
         self._predict_from_features_task = None
+        self.__callbacks: Dict[str, Callable] = {}
```

Comment:
we'll potentially want to add callbacks for the `trainer` and `evaluator`, so let's rename this to `__predict_callbacks`